api/v1alpha1/virtualmachine_conversion.go

```diff
@@ -892,13 +892,150 @@ func restore_v1alpha2_VirtualMachineReadinessProbeSpec(
+	dst.Status.Conditions = append(dst.Status.Conditions[:preReqCondIdx], dst.Status.Conditions[preReqCondIdx+1:]...)
+
+	// If any of the v1a2 conditions were already evaluated, v1a1 PrereqReady condition has been removed, so move on.
+	if vmClassCond != nil || vmImageCond != nil || vmSetResourcePolicy != nil || vmBootstrap != nil {
+		return dst.Status.Conditions
+	}
+
+	var conditions []metav1.Condition
+	// If we don't have any of the new v1a2 conditions, use the v1a1 PrereqReady condition to fill
+	// that in. This means that this was originally a v1a1 VM.
+	if preReqCond.Status == metav1.ConditionTrue {
+
+		// The class and image are always required fields.
+		conditions = append(conditions, metav1.Condition{
+			Type:               v1alpha2.VirtualMachineConditionClassReady,
+			Status:             metav1.ConditionTrue,
+			LastTransitionTime: preReqCond.LastTransitionTime,
+			Reason:             string(metav1.ConditionTrue),
+		})
+		conditions = append(conditions, metav1.Condition{
+			Type:               v1alpha2.VirtualMachineConditionImageReady,
+			Status:             metav1.ConditionTrue,
+			LastTransitionTime: preReqCond.LastTransitionTime,
+			Reason:             string(metav1.ConditionTrue),
+		})
+
+		if dst.Spec.Reserved != nil && dst.Spec.Reserved.ResourcePolicyName != "" {
+			conditions = append(conditions, metav1.Condition{
+				Type:               v1alpha2.VirtualMachineConditionVMSetResourcePolicyReady,
+				Status:             metav1.ConditionTrue,
+				LastTransitionTime: preReqCond.LastTransitionTime,
+				Reason:             string(metav1.ConditionTrue),
+			})
+		}
+		if dst.Spec.Bootstrap != nil {
+			conditions = append(conditions, metav1.Condition{
+				Type:               v1alpha2.VirtualMachineConditionBootstrapReady,
+				Status:             metav1.ConditionTrue,
+				LastTransitionTime: preReqCond.LastTransitionTime,
+				Reason:             string(metav1.ConditionTrue),
+			})
+		}
+
+	} else if preReqCond.Status == metav1.ConditionFalse {
+		// Infer what v1a2 conditions need to be true/false from preReq reason.
+		// Order of evaluation of objects for v1a1 PreReq Condition:
+		// 1. VM Class then
+		// 2. VM Image
+
+		if preReqCond.Reason == VirtualMachineClassNotFoundReason {
+			conditions = append(conditions, metav1.Condition{
+				Type:               v1alpha2.VirtualMachineConditionClassReady,
+				Status:             metav1.ConditionFalse,
+				LastTransitionTime: preReqCond.LastTransitionTime,
+				Reason:             string(metav1.ConditionFalse),
+			})
+			// v1a1 Image preReq hasn't been evaluated yet -- mark unknown
+			conditions = append(conditions, metav1.Condition{
+				Type:               v1alpha2.VirtualMachineConditionImageReady,
+				Status:             metav1.ConditionUnknown,
+				LastTransitionTime: preReqCond.LastTransitionTime,
+				Reason:             string(metav1.ConditionUnknown),
+			})
+		} else if preReqCond.Reason == VirtualMachineImageNotFoundReason {
```

Comment:
Similarly too can we include the Content* and VirtualMachineImageNotReadyReason reason?